lib/knockoff/config.rb

```diff
@@ -66,58 +66,17 @@ def parse_knockoff_replica_envs_to_configs
       # and don't add the uri to the final list if it can't be parsed
       replica_env_keys.map.with_index(0) do |env_key, index|
         begin
-          uri = URI.parse(ENV[env_key])
 
           # Configure parameters such as prepared_statements, pool, reaping_frequency for all replicas.
-          replica_config = ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas')&.first || {}
-
-          adapter =
-            if uri.scheme == "postgres"
-              'postgresql'
-            else
-              uri.scheme
-            end
-
-          # Base config from the ENV uri. Sqlite is a special case
-          # and all others follow 'normal' config
-          uri_config =
-            if uri.scheme == 'sqlite3'
-              {
-                'adapter' => adapter,
-                'database' => uri.to_s.split(':')[1]
-              }
-            else
-              {
-                'adapter' => adapter,
-                'database' => (uri.path || "").split("/")[1],
-                'username' => uri.user,
-                'password' => uri.password,
-                'host' => uri.host,
-                'port' => uri.port
-              }
-            end
-
-          # Store the hash in configuration and use it when we establish the connection later.
-          # TODO: In ActiveRecord >= 6, this is a deprecated way to set a configuration. However
-          # there appears to be an issue when calling `ActiveRecord::Base.configurations.to_h` in
-          # version 6.0.4.8 where
-          # multi-database setup is being ignored / dropped. For example if a database.yml setup
-          # has something like..
-          #
-          # development:
-          #   primary:
-          #     ...
-          #   other:
-          #     ...
-          #
-          # then the 'other' database configuration is being dropped.
-          key = "knockoff_replica_#{index}"
-          config = replica_config.configuration_hash.symbolize_keys.merge(uri_config)
-          env_name = ActiveRecord::Base.configurations.configurations.first.env_name
-          new_config = ActiveRecord::DatabaseConfigurations::HashConfig.new(env_name, key, config)
-          ActiveRecord::Base.configurations.configurations << new_config
-
-          @replicas_configurations[key] = config
+          ActiveRecord::Base.configurations.configs_for(env_name: 'knockoff_replicas').each do |it|
+            register_replica_copy(index, env_key, it.configuration_hash)
+            break
```

Comment:
This is obviously silly, but I don't really understand why it was just taking the first one? This would create a new db connection each time this is called, but the old ones wouldn't get cleaned up. Maybe it was fine because it was being set using ` ActiveRecord::Base.configurations[key]` which is not longer available (See next comment).